### PR TITLE
Support hdf5 1.14.0

### PR DESCRIPTION
**Pull request by mulimoen**
The work done to support 1.13.0 means 1.14.0 is trivial to support

### CODE REVIEW

**Comment by jan-kaspar** [comment]
Hi all, what is the status of this PR? I'd be very grateful for the support of 1.14...

BTW: could one ignore the "patch" part here: https://github.com/aldanor/hdf5-rust/blob/master/hdf5-sys/build.rs#L32 ? For example on ArchLinux there is just a number after the dash, https://archlinux.org/packages/community/x86_64/hdf5/, thus failing the regex.

Many thanks in advance!

**Comment by aldanor** [comment]
@mulimoen I'll take a look at this (and other PRs) on the weekend, is h5d the only change here?

**Comment by mulimoen** [comment]
@aldanor Yes, everything else is covered by the support for 1.13

We should probably remove support for 1.13 at some point considering odd minor versions are "test" versions

**Comment by mulimoen** [comment]
Patched versions in e.g. Arch linux currently has the version number "1.14.1-2" which should now be supported

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -193,7 +197,7 @@ jobs:
       fail-fast: false
       matrix:
         rust: [stable]
-        version: ["1.8", "1.10", "1.12", "1.13"]
+        version: ["1.8", "1.10", "1.12", "1.14"]
```
Comment (aldanor) [inline .github/workflows/ci.yml:200]: Yea, probably the right call to just throw away 1.13 from the CI and replace it with 1.14 where possible.

**Comment by aldanor** [comment]
Looks good, thanks! Unless there's anything else to be done here I'll merge this so we can proceed forward.

**Comment by aldanor** [comment]
Just double checking - would this close https://github.com/aldanor/hdf5-rust/issues/230?

**Comment by mulimoen** [comment]
Feel free to merge. I'll prepare something to rid us of 1.13 later

**Review by aldanor — APPROVED** [review]

**Pull request merged**